### PR TITLE
chore: introducing check for number of outcomes to enable cross matching

### PR DESCRIPTION
**Pull request by pawel-betdex**
The cross-matching code is written in a way that can handle markets with any number of outcomes. However, the more outcomes there is the more compute units it requires and the worry that transactions start to fail.

So done some testing and looks like at about 7 outcomes standard 20k of compute units is no longer enough on occasion to create orders with cross matching enabled (8 outcomes is 100% failure).

Based on that then I've decided to limit number of outcomes allowed for cross-matching market to 5.

### CODE REVIEW

File: tests/util/wrappers.ts [hunk 1]
```diff
@@ -1060,14 +1065,18 @@ export class MonacoMarket {
...
+    do {
       const orderPk = await this.processNextOrderRequest();
-      orderPks.push(orderPk);
-    }
+      if (orderPk === null) {
+        tryMore--;
+      } else {
+        orderPks.push(orderPk);
+      }
+    } while (tryMore > 0);
```
Comment (obi-ewan) [inline tests/util/wrappers.ts:1079]: why is this needed now? is there a delay between creating order requests and them being visible on the queue with the local validator?

**Review by obi-ewan — APPROVED** [review]

File: tests/util/wrappers.ts (hunk 1 above)
Comment (pawel-betdex) [reply tests/util/wrappers.ts:1079]: pretty much what was my experience. but even before that the loop had an issue as `processNextOrderRequest` is working with its own freshly fetched copy of the queue. so decided to get rid of the top level fetch.

**Review by druspencer — APPROVED** [review]
